### PR TITLE
Cleanup documentation

### PR DESCRIPTION
**Pull request by adonagy**
- Moved the md files under docs
- Cherry-picked the commit editing the README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -137,3 +137,30 @@ Open your browser and navigate to [http://localhost:4200](http://localhost:4200)
...
+  - [P2P communication](https://github.com/openmina/openmina/blob/documentation/docs/p2p_service.md)
+    - [GossipSub](https://github.com/openmina/mina-wiki/blob/3ea9041e52fb2e606918f6c60bd3a32b8652f016/p2p/mina-gossip.md)
+  - [Block production](https://github.com/JanSlobodnik/pre-publishing/blob/main/block-production.md)
+  - [Scan state](https://github.com/JanSlobodnik/pre-publishing/blob/main/scan-state.md)
+  - [SNARKs](https://github.com/JanSlobodnik/pre-publishing/blob/main/SNARKs.md)
+  - Developer tools
+    - [Debugger](https://github.com/openmina/mina-network-debugger/blob/main/README.md)
+  - [Front End](https://github.com/openmina/mina-frontend/blob/main/README.md)
+    - [Dashboard](https://github.com/openmina/mina-frontend/blob/main/docs/MetricsTracing.md#Dashboard)
+- [Testing](https://github.com/openmina/openmina/blob/develop/testing.md)
```
Comment (adonagy) [inline README.md:158]: @JanSlobodnik You can link files that are part of the repository using relative paths. For example, this line could be:

```
[Testing](docs/testing/testing.md)
```

File: README.md [hunk 2]
```diff
@@ -137,3 +137,30 @@ Open your browser and navigate to [http://localhost:4200](http://localhost:4200)
 ---
 
 [Details regarding architecture](ARCHITECTURE.md)
+
+
+# The Open Mina documentation overview
```
Comment (adonagy) [inline README.md:142]: Maybe move this to the top of the README? @tizoc any suggestions?

Comment (JanSlobodnik) [reply README.md:142]: Juraj wants it at the bottom of the readme

File: README.md [hunk 3]
```diff
@@ -137,3 +137,30 @@ Open your browser and navigate to [http://localhost:4200](http://localhost:4200)
...
+
+- [Why we are developing Open Mina](https://github.com/JanSlobodnik/pre-publishing/blob/main/why-openmina.md)
+- What is Open Mina?
+  - [Openmina Node](https://github.com/openmina/openmina/blob/main/README.md)
+  - [The Mina Web Node](https://github.com/openmina/webnode/blob/main/README.md)
+- Core components
+  - [P2P communication](https://github.com/openmina/openmina/blob/documentation/docs/p2p_service.md)
+    - [GossipSub](https://github.com/openmina/mina-wiki/blob/3ea9041e52fb2e606918f6c60bd3a32b8652f016/p2p/mina-gossip.md)
+  - [Block production](https://github.com/JanSlobodnik/pre-publishing/blob/main/block-production.md)
+  - [Scan state](https://github.com/JanSlobodnik/pre-publishing/blob/main/scan-state.md)
```
Comment (tizoc) [inline README.md:152]: some of these docs are now in this repository right? these links need to be updated (using relative paths like @adonagy suggested bellow)

**Comment by adonagy** [comment]
Just noticed that we are referencing https://github.com/openmina/openmina/blob/documentation/docs/p2p_service.md. The `documentation` branch is not merged yet. 
@tizoc We can leave the link as is, or can we merge the branch @dkuehr @JanSlobodnik ?

**Comment by tizoc** [comment]
@adonagy I mentioned that to @JanSlobodnik earlier today, I think that for this release it is not a big deal, but in general we have to sort that out because we have documentation referenced here that spread across different branches/repos.

File: docs/snark-work.md [hunk 4]
```diff
@@ -115,35 +109,39 @@ After this, the updated SNARK pool with the completed (but not yet included in a
...
-
-![image](https://github.com/JanSlobodnik/pre-publishing/assets/60480123/e9067fcc-2180-46b3-b445-a03ac77c2b16)
+<img width="318" alt="image" src="https://github.com/openmina/openmina/assets/60480123/46ec6804-b767-4054-aaeb-0287ab9cda09">
 
 
 ### Receiving SNARK from Rust node
 
+Rust node sends SNARK via P2P.
 
+<img width="342" alt="image" src="https://github.com/openmina/openmina/assets/60480123/067de8a5-246e-4b59-a85b-7f2393cc19c3">
```
Comment (tizoc) [inline docs/snark-work.md:119]: @JanSlobodnik was the switch from markdown image tags to these html tags (with fixed width) intentional?

**Pull request merged**